float32 x
float32 y
bool grasping
string current_grasp
